crates/extensions/c8y_mapper_ext/src/config_operations.rs

```diff
@@ -151,44 +155,37 @@ impl CumulocityConverter {
                 let event_response_id = self.http_proxy.send_event(create_event).await?;
 
                 // Send a request to the Uploader to upload the file asynchronously.
-                let uploaded_file_path = self
-                    .config
-                    .data_dir
-                    .file_transfer_dir()
-                    .join(target.external_id.as_ref())
-                    .join("config_snapshot")
-                    .join(format!(
-                        "{}-{}",
-                        response.config_type.replace('/', ":"),
-                        cmd_id
-                    ));
-
-                let binary_upload_event_url = self
-                    .c8y_endpoint
-                    .get_url_for_event_binary_upload_unchecked(&event_response_id);
-
-                let upload_request = UploadRequest::new(
-                    self.auth_proxy
-                        .proxy_url(binary_upload_event_url.clone())
-                        .as_str(),
-                    &uploaded_file_path,
-                );
+                let config_filename =
+                    format!("{}-{}", response.config_type.replace('/', ":"), cmd_id);
 
-                self.uploader_sender
-                    .send((cmd_id.into(), upload_request))
-                    .await
-                    .map_err(CumulocityMapperError::ChannelError)?;
+                let uploaded_file_url = format!(
+                    "http://{}/tedge/file-transfer/{external_id}/config_snapshot/{config_filename}",
+                    &self.config.tedge_http_host,
+                    external_id = target.external_id.as_ref()
+                );
 
-                self.pending_upload_operations.insert(
+                self.pending_fts_download_operations.insert(
                     cmd_id.into(),
-                    UploadOperationData {
+                    FtsDownloadOperationData {
+                        url: uploaded_file_url.clone(),
+                        download_type: FtsDownloadOperationType::ConfigDownload,
                         smartrest_topic,
                         clear_cmd_topic: message.topic.clone(),
-                        c8y_binary_url: binary_upload_event_url.to_string(),
-                        operation: CumulocitySupportedOperations::C8yUploadConfigFile,
+                        event_response_id,
                     },
                 );
 
+                let destination_path = self.config.data_dir.cache_dir().join(config_filename);
```

Comment:
Using `cache_dir` to store a temporary file! This needs to be discussed.

1. The `cache_dir` aka `/var/tedge/cache` is associated with tedge-agent's file transfer service. Here is tedge-mapper, may be different file system. 
2. The original purpose of `cache_dir` is to prevent multiple downloads for the same file from cloud. For example, firmware_update, you want to apply the same firmware to multiple child devices. The usage of `cache_dir` here doesn't suit the purpose.

For me, a file uploaded from a device per operation is just a temporary file. I think using `/tmp` is better, but I'm opened for more opinions.

Reply:
Agreed, this should definitely be a tmpdir, I got a little lazy and tried to find something similar enough in the `DataDir` object, but cache directory is not it. `TempTedgeDir` is close, but because it uses `TempDir`, it will get removed when it gets out of scope. I'll either use `tmp.path` directly and manually remove the file when finishing the operation handling, or add `TempTedgeDir` to the `pending_fts_download_operations` hashmap, so it gets removed automatically.